server/sonar-server/src/main/java/org/sonar/server/issue/IssueService.java

```diff
@@ -68,10 +64,10 @@ public void assign(String issueKey, @Nullable String assignee) {
     DbSession session = dbClient.openSession(false);
     try {
       DefaultIssue issue = issueFinder.getByKey(session, issueKey).toDefaultIssue();
-      User user = null;
+      UserDto user = null;
       if (!Strings.isNullOrEmpty(assignee)) {
-        user = userFinder.findByLogin(assignee);
-        checkRequest(user != null, "Unknown user: %s", assignee);
+        user = dbClient.userDao().selectByLogin(session, assignee);
```

Comment:
In fact there's already, but I think it's better for the "client" to filter active/inactive users